include/pid-controller-api.h

```diff
@@ -10,12 +10,16 @@
  * parameters and integrates the error history for advanced control functionality.
  */
 
-#ifndef _PID_CONTROLLER_API_H_
-#define _PID_CONTROLLER_API_H_
+#ifndef PID_CONTROLLER_API_H
+#define PID_CONTROLLER_API_H
 
 #include <arena-allocator-api.h>
 #include "pid-controller.h"
 
+#define PID_SUCCESS 0
+#define PID_ERROR_NULL_PTR -1
+#define PID_ERROR_MEM_ALLOC -2
```

Comment:
Use an `enum` instead of multiple defines too keep the code simple to manage and understand.
```suggestion
typedef enum _PidReturnCode_t {
    PID_OK = 0,
    PID_NULL_POINTER,
    PID_ALLOCATION_ERROR
} PidReturnCode_t;
```